airflow/ti_deps/deps/ready_to_reschedule.py

```diff
@@ -52,14 +52,14 @@ def _get_dep_statuses(self, ti, session, dep_context):
                 reason="The task instance is not in State_UP_FOR_RESCHEDULE or NONE state.")
             return
 
-        task_reschedules = TaskReschedule.find_for_task_instance(task_instance=ti)
-        if not task_reschedules:
+        task_reschedule = TaskReschedule.query_for_task_instance(task_instance=ti, descending=True).first()
```

Comment:
Please pass on the same session object.
```suggestion
        task_reschedule = (
            TaskReschedule.query_for_task_instance(task_instance=ti, descending=True, session=session)
            .with_entities(TR.reschedule_date)
            .first()
        )
```

Reply:
You missed Kamil's suggestion of `.with_entites` here.

Reply:
I think it's there :)